packages/stokhos/src/sacado/kokkos/pce/Stokhos_Sacado_Kokkos_UQ_PCE.hpp

```diff
@@ -12,6 +12,8 @@
 
 #include "Stokhos_ConfigDefs.h"
 
+#warning "The PCE scalar type is deprecated."
```

Comment:
In general, it is better to deprecate actual C++ types, functions, etc.  Just deprecating the entire header file is not ideal.  See:

* https://tribitspub.github.io/TriBITS/users_guide/index.html#regulated-backward-compatibility-and-deprecated-code

Reply:
Everything in this header, and all of the code that includes this header will be deprecated, so I don't see a reason to put the macro on every single class/function (which would be hundreds of functions/classes).